### PR TITLE
grpc_cli: init at 1.46.6

### PR DESCRIPTION
**Pull request by doriath**
###### Description of changes

The command line tool for interacting with grpc services.

https://github.com/grpc/grpc/blob/master/doc/command_line_tool.md

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -3725,6 +3725,12 @@
     githubId = 39825;
     name = "Dominik Honnef";
   };
+  doriath = {
```
Comment (onny) [inline maintainers/maintainer-list.nix:3728]: Please put these changes into a separate commit with the title like "maintainers: add doriath"

File: pkgs/tools/networking/grpc_cli/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+stdenv.mkDerivation rec {
+  pname = "grpc_cli";
+  version = "1.46.6";
+
+  buildInputs = [ automake cmake autoconf curl numactl ];
+  src = fetchFromGitHub {
+    owner = "grpc";
+    repo = "grpc";
+    rev = "18dda3c586b2607d8daead6b97922e59d867bb7d";
+    sha256 = "sha256-m55FKwqlAQ2Ge7y4RMFO4W1onu0NYARWrOygChJMxng=";
```
Comment (onny) [inline pkgs/tools/networking/grpc_cli/default.nix:12]:
```suggestion
    hash = "sha256-m55FKwqlAQ2Ge7y4RMFO4W1onu0NYARWrOygChJMxng=";
```

File: pkgs/tools/networking/grpc_cli/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "grpc_cli";
+  version = "1.46.6";
+
+  buildInputs = [ automake cmake autoconf curl numactl ];
+  src = fetchFromGitHub {
+    owner = "grpc";
+    repo = "grpc";
+    rev = "18dda3c586b2607d8daead6b97922e59d867bb7d";
```
Comment (onny) [inline pkgs/tools/networking/grpc_cli/default.nix:11]:
```suggestion
    rev = "v${version}";
```

File: pkgs/tools/networking/grpc_cli/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
+{ lib, stdenv, fetchFromGitHub, automake, cmake, autoconf, curl, numactl }:
+
+stdenv.mkDerivation rec {
+  pname = "grpc_cli";
+  version = "1.46.6";
+
+  buildInputs = [ automake cmake autoconf curl numactl ];
```
Comment (onny) [inline pkgs/tools/networking/grpc_cli/default.nix:7]: Please put buildInputs after src

File: pkgs/tools/networking/grpc_cli/default.nix [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  src = fetchFromGitHub {
+    owner = "grpc";
+    repo = "grpc";
+    rev = "18dda3c586b2607d8daead6b97922e59d867bb7d";
+    sha256 = "sha256-m55FKwqlAQ2Ge7y4RMFO4W1onu0NYARWrOygChJMxng=";
+    fetchSubmodules = true;
+  };
+  configurePhase = "true";
```
Comment (onny) [inline pkgs/tools/networking/grpc_cli/default.nix:15]: Isn't this true by default?

File: pkgs/tools/networking/grpc_cli/default.nix [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  buildPhase = ''
+    mkdir -p $out/bin
+    mkdir -p cmake/build
+    cd cmake/build
+    cmake -DgRPC_BUILD_TESTS=ON ../..
+    make grpc_cli
+    cp grpc_cli $out/bin
```
Comment (onny) [inline pkgs/tools/networking/grpc_cli/default.nix:22]: This should probably belong to installPhase?

File: pkgs/tools/networking/grpc_cli/default.nix [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+  '';
+  installPhase = "true";
+
+  meta = with lib; {
+    description = "The command line tool for interacting with grpc services.";
+    homepage = "https://github.com/grpc/grpc/blob/master/doc/command_line_tool.md";
```
Comment (onny) [inline pkgs/tools/networking/grpc_cli/default.nix:28]:
```suggestion
    homepage = "https://github.com/grpc/grpc";
```

File: pkgs/tools/networking/grpc_cli/default.nix (hunk 5 above)
Comment (doriath) [reply pkgs/tools/networking/grpc_cli/default.nix:15]: I believe by default it calls `./configure`, based on https://nixos.org/manual/nixpkgs/stable/#ssec-configure-phase. And I want this phase to be a no-op, so I just run `true`.

File: pkgs/tools/networking/grpc_cli/default.nix (hunk 4 above)
Comment (doriath) [reply pkgs/tools/networking/grpc_cli/default.nix:7]: Thank you for suggestion, done.

File: pkgs/tools/networking/grpc_cli/default.nix (hunk 6 above)
Comment (doriath) [reply pkgs/tools/networking/grpc_cli/default.nix:22]: Good idea, moved to install phase.

File: pkgs/tools/networking/grpc_cli/default.nix (hunk 3 above)
Comment (doriath) [reply pkgs/tools/networking/grpc_cli/default.nix:11]: Done.

File: pkgs/tools/networking/grpc_cli/default.nix (hunk 2 above)
Comment (doriath) [reply pkgs/tools/networking/grpc_cli/default.nix:12]: Done.

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (doriath) [reply maintainers/maintainer-list.nix:3728]: Moved to separate commit.

File: pkgs/tools/networking/grpc_cli/default.nix (hunk 7 above)
Comment (doriath) [reply pkgs/tools/networking/grpc_cli/default.nix]: Thank you for all the comments!

File: pkgs/tools/networking/grpc_cli/default.nix [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+  version = "1.46.6";
+
+  src = fetchFromGitHub {
+    owner = "grpc";
+    repo = "grpc";
+    rev = "v${version}";
+    hash = "sha256-m55FKwqlAQ2Ge7y4RMFO4W1onu0NYARWrOygChJMxng=";
+    fetchSubmodules = true;
+  };
+  buildInputs = [ automake cmake autoconf curl numactl ];
```
Comment (NickCao) [inline pkgs/tools/networking/grpc_cli/default.nix:14]: Some of these are apparently nativeBuildInputs (which is for compilers/build-time tools), buildInputs is only for libraries (run-time).

File: pkgs/tools/networking/grpc_cli/default.nix [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+  configurePhase = "true";
+  buildPhase = ''
+    mkdir -p $out/bin
+    mkdir -p cmake/build
+    cd cmake/build
+    cmake -DgRPC_BUILD_TESTS=ON ../..
+    make grpc_cli
+  '';
```
Comment (NickCao) [inline pkgs/tools/networking/grpc_cli/default.nix:22]:
```suggestion
  makeFlags = [ "grpc_cli" ];
```

Also remove the configurePhase override above.

File: pkgs/tools/networking/grpc_cli/default.nix [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+  installPhase = "cp grpc_cli $out/bin";
```
Comment (NickCao) [inline pkgs/tools/networking/grpc_cli/default.nix:23]:
```suggestion
  installPhase = ''
    runHook preInstall
 
    install -Dm555 grpc_cli "$out/bin/grpc_cli"
    
    runHook postInstall
  '';
```

File: pkgs/tools/networking/grpc_cli/default.nix (hunk 8 above)
Comment (doriath) [reply pkgs/tools/networking/grpc_cli/default.nix]: Moved automake, cmake and autoconfg to nativeBuildInputs.

File: pkgs/tools/networking/grpc_cli/default.nix (hunk 9 above)
Comment (doriath) [reply pkgs/tools/networking/grpc_cli/default.nix]: I also added cmakeFlags and it worked. Thank you!

File: pkgs/tools/networking/grpc_cli/default.nix (hunk 10 above)
Comment (doriath) [reply pkgs/tools/networking/grpc_cli/default.nix]: Applied, thank you!

**Comment by NickCao** [comment]
Build is failing due to hash mismatch:
```
error: hash mismatch in fixed-output derivation '/nix/store/lc8niax8sggk6q3fxzpi01sf7j5h2d27-source.drv':
         specified: sha256-m55FKwqlAQ2Ge7y4RMFO4W1onu0NYARWrOygChJMxng=
            got:    sha256-UPenQh6+FBryQiOoeijsXkCZjlMzYljkg2aUtSFJFL4=
```

**Comment by NickCao** [comment]
Still failing due to `error: format not a string literal and no format arguments [-Werror=format-security]`, may have to disable some hardening.

File: pkgs/tools/networking/grpc_cli/default.nix [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+    install -Dm555 grpc_cli "$out/bin/grpc_cli"
+
+    runHook postInstall
+  '';
+  meta = with lib; {
+    description = "The command line tool for interacting with grpc services.";
+    homepage = "https://github.com/grpc/grpc";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ doriath ];
+    platforms = [ "x86_64-linux" "x86_64-darwin" ];
```
Comment (NickCao) [inline pkgs/tools/networking/grpc_cli/default.nix:29]: It does not evaluate on x86_64-darwin, and I think it might build on aarch64-linux. How was these platforms determined?

**Review by NickCao — COMMENTED** [review]
Other than questions about the platforms, LGTM.

File: pkgs/tools/networking/grpc_cli/default.nix (hunk 11 above)
Comment (doriath) [reply pkgs/tools/networking/grpc_cli/default.nix]: Indeed, x86 and aarch on linux worked, and removed the darwin one. Thank you!.

File: pkgs/tools/networking/grpc_cli/default.nix [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+    install -Dm555 grpc_cli "$out/bin/grpc_cli"
+
+    runHook postInstall
+  '';
+  meta = with lib; {
+    description = "The command line tool for interacting with grpc services.";
+    homepage = "https://github.com/grpc/grpc";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ doriath ];
+    platforms = [ "x86_64-linux" "aarch64-darwin" ];
```
Comment (NickCao) [inline pkgs/tools/networking/grpc_cli/default.nix:29]: typo? darwin is still here. You can use platforms.linux to denote all linux platforms.

Comment (doriath) [reply pkgs/tools/networking/grpc_cli/default.nix]: Thank you, did not know that platforms.linux could be used. Is there a way to test / simulate other platforms (to debug the issue and try to fix them in followup PR?)

Comment (NickCao) [reply pkgs/tools/networking/grpc_cli/default.nix]: Nothing better than spinning up VMs.

Comment (doriath) [reply pkgs/tools/networking/grpc_cli/default.nix]: Unfortunately it seems that aarch64-linux fails. Will keep `platforms = [ "x86_64-linux" ];` that works.

**Review by NickCao — APPROVED** [review]
